packages/advanced-logic/src/extensions/payment-network/declarative.ts

```diff
@@ -203,11 +205,14 @@ export default class DeclarativePaymentNetwork<
     actionSigner: IdentityTypes.IIdentity,
     timestamp: number,
   ): ExtensionTypes.IState {
-    if (!requestState.payer) {
-      throw Error(`The request must have a payer`);
+    if (!requestState.payer && !extensionState.values.thirdparty) {
+      throw Error(`The request must have a payer or a thirdparty`);
```

Comment:
I'm a bit surprised by this condition, a declarative request allows the third party to detect that the payment has been made, which is an action done by the issuer, not the payer. Am I missing something?

Reply:
@yomarion this action is a "payment sent" action

Reply:
Yes, I know. I think there are flows, it's not because the issuer trusts the 3rd party that the payer has to trust it.